README.md

```diff
@@ -24,6 +24,129 @@ def deps do
+
+```elixir
+config :soroban, :http_client_impl, YourApp.CustomClientImpl
+```
+
+### Custom JSON library
+
+Following the same approach as the HTTP client, the JSON parsing library can also be configured. Defaults to [`Jason`][jason_url].
+
+```elixir
+config :soroban, :json_library, YourApp.CustomJSONLibrary
+```
+
+## Soroban RPC endpoints
+
+### Simulate Transaction
+
+Submit a trial contract invocation to get back return values, expected ledger footprint, and expected costs.
+
+**Parameters**
+
+- `base64_envelope`: The transaction to be simulated (serialized in base64).
+
+**Example**
+
+```elixir
+base64_envelope =
+  "AAAAAgAAAADWKIRtrzg/aTCtUHeZnpyYu0iNxJxcn4tr0jXG2hOIlwAAAGQABzbWAAAAAwAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAGAAAAAAAAAAEAAAADQAAACC8xDySpTRgcsckFZY9QBvIP3LL70Jp0xG3cmpCvp0d/QAAAA8AAAAJaW5jcmVtZW50AAAAAAAAEwAAAAAAAAAA1iiEba84P2kwrVB3mZ6cmLtIjcScXJ+La9I1xtoTiJcAAAADAAAACgAAAAAAAAAAAAAAAQAAAAC8xDySpTRgcsckFZY9QBvIP3LL70Jp0xG3cmpCvp0d/QAAAAlpbmNyZW1lbnQAAAAAAAACAAAAEwAAAAAAAAAA1iiEba84P2kwrVB3mZ6cmLtIjcScXJ+La9I1xtoTiJcAAAADAAAACgAAAAAAAAAAAAAAAAAAAAA="
+
+{:ok,
+ %Soroban.RPC.SimulateTransactionResponse{
+   results: [
+     %{
+       auth: nil,
+       events: nil,
+       footprint:
+         "AAAAAgAAAAYU0EuZrCKggMgcYHtwMuiHqnrYwhksO17kfjwJ8h2l3QAAABQAAAAHCoKrtqgxTcxBJ+F9JX+3Gvlw3NtYGwCu8hzxUsbupwIAAAAA",
+       xdr: "AAAAEAAAAAEAAAACAAAADwAAAAVIZWxsbwAAAAAAAA8AAAAFd29ybGQAAAA="
+     }
+   ],
+   cost: %{cpu_insns: "1048713", mem_bytes: "1201148"},
+   latest_ledger: "475528",
+   error: nil
+ }} = Soroban.RPC.simulate_transaction(base64_envelope)
+
+```
+
+### Send Transaction
+
+Submit a real transaction to the Stellar network. This is the only way to make changes "on-chain".
+
+Unlike Horizon, this does not wait for transaction completion. It simply validates and enqueues the transaction. Clients should call [Get Transaction](#get-transaction) to learn about transaction success/failure.
+
+This supports all transactions, not only smart contract-related transactions.
+
+**Parameters**
+
+- `base64_envelope`: The signed transaction to broadcast (serialized in base64).
+
+**Example**
+
+```elixir
+base64_envelope =
+  "AAAAAgAAAADBPp7TMinJylnn+6dQXJACNc15LF+aJ2Py1BaR4P10JAAAAGQAAFBfAAAAIQAAAAEAAAAAAAAAAAAAAABkUtg3AAAAAAAAAAEAAAABAAAAAME+ntMyKcnKWef7p1BckAI1zXksX5onY/LUFpHg/XQkAAAAGAAAAAAAAAADAAAADQAAACAU0EuZrCKggMgcYHtwMuiHqnrYwhksO17kfjwJ8h2l3QAAAA8AAAAFaGVsbG8AAAAAAAAPAAAABXdvcmxkAAAAAAAAAgAAAAYU0EuZrCKggMgcYHtwMuiHqnrYwhksO17kfjwJ8h2l3QAAABQAAAAHCoKrtqgxTcxBJ+F9JX+3Gvlw3NtYGwCu8hzxUsbupwIAAAAAAAAAAAAAAAAAAAAB4P10JAAAAEDS4+hvSG1JqhOIPaGSqUerNsjhIcS+AwWhH/K8IOafcmMlZJoyZvMftV1QcdWA/LQhr2QJRTWNT6I52/eTP3IH"
+
+{:ok,
+  %Soroban.RPC.SendTransactionResponse{
+    status: "PENDING",
+    hash: "a4721e2a61e9a6b3f54030396e41c3e352101e6cd649b4453e89fb3e827744f4",
+    latest_ledger: "476420",
+    latest_ledger_close_time: "1683150612",
+    error_result_xdr: nil
+  }} = Soroban.RPC.send_transaction(base64_envelope)
```

Comment:
What do you guys think about refactoring in this way:

```elixir
Soroban.RPC.send_transaction(base64_envelope)

{:ok,
 %Soroban.RPC.SendTransactionResponse{
   status: "PENDING",
   hash: "a4721e2a61e9a6b3f54030396e41c3e352101e6cd649b4453e89fb3e827744f4",
   latest_ledger: "476420",
   latest_ledger_close_time: "1683150612",
   error_result_xdr: nil
 }}
```

It could be better if anyone wants to copy-paste into a terminal... instead of the pattern matching approach. 

If you agree, please update for every endpoint